lib/segment/src/types.rs

```diff
@@ -545,6 +549,63 @@ pub enum PayloadSchemaType {
     Integer,
     Float,
     Geo,
+    Text,
+}
+
+/// Payload type with parameters
+#[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq, Hash, Eq)]
+#[serde(rename_all = "snake_case")]
+#[serde(tag = "type")]
+pub enum PayloadSchemaParams {
```

Comment:
I propose to move these structs to `src/data_types` directory

Reply:
WIll do in another PR

Reply:
okay